### DISCUSSION TITLE
Unable to find the requested cms layout '1' for nav item page id '1

### DISCUSSION THREAD

**Discussion by kangzhaolong** [luyadev/luya#2157] [Questions & Problems]
Hello 
I create the luya-kickstarter-bootstrap4 on my pc and config well, and I upload to git server.

on my cloud server, I get the package from git server and do the "composer update" , THEN I EXECUTE THE COMMAND  "./vender/bin/luya import", **It seams that the "id"  of database cms_layout will be increase automatically .** 
BUT in the cms/frontend/default page, it still search the layout id which value is 1.

then I visit the default cms, (e.g. https://xxxx/cms), The default page will show below information:

Invalid Configuration – [yii\base\InvalidConfigException](http://www.yiiframework.com/doc-2.0/yii-base-invalidconfigexception.html)
Unable to find the requested cms layout '1' for nav item page id '1'. Make sure your page does not have an old inactive/deleted cms layout selected.

1. in /xxxxx/cms/vendor/luyadev/luya-module-cms/src/models/NavItemPage.phpat line 161
152153154155156157158159160161162163164165166167168169170    

```
    public function getContent()
    {
        if (!$this->layout) {
            throw new InvalidConfigException("Unable to find the requested cms layout '{$this->layout_id}' for nav item page id '{$this->id}'. Make sure your page does not have an old inactive/deleted cms layout selected.");
        }
 
        $placholders = [];
        foreach ($this->layout->getJsonConfig('placeholders') as $row) {
            foreach ($row as $item) {
                $placholders[$item['var']] = $this->renderPlaceholder($item['var']);
            }
        }
 ```
2. in /xxxx/cms/vendor/luyadev/luya-module-cms/src/frontend/base/Controller.php at line 161– luya\cms\models\NavItemPage::getContent()
155156157158159160161162163164165166167 

```
        $typeModel->setOptions([
            'navItemId' => $navItemId,
            'restString' => $appendix,
        ]);
 
        $content = $typeModel->getContent();
 
        Yii::$app->trigger(self::EVENT_RENDER_CMS_PAGE, new Event(['data' => ['navItemId' => $navItemId]]));
 
        if ($content instanceof Response) {
            return Yii::$app->end(0, $content);
        }
```

I am thinking if there is some problem in "import process" or the cms default page for using the id of cms_layout?

**Comment by nadar** [comment]
What version of luya, cms and admin are you using? What database server do you use, mysql?

**Reply by kangzhaolong** [reply]
Installing dependencies from lock file (including require-dev)
Package operations: 8 installs, 0 updates, 0 removals
  - Installing luyadev/luya-composer (1.1.2): Extracting archive
  - Installing luyadev/yii-helpers (1.2.2): Extracting archive
  - Installing luyadev/luya-core (2.2.0): Extracting archive
Package luyadev/luya-core will be ignored by luyadev installer as its not part of the composer.json requirements.
  - Installing luyadev/luya-bootstrap4 (1.1.2): Extracting archive
  - Installing luyadev/luya-deployer (1.0.9): Extracting archive
  - Installing luyadev/luya-generic (1.1.2): Extracting archive
  - Installing luyadev/luya-module-admin (4.5.0): Extracting archive
  - Installing luyadev/luya-module-cms (4.3.0): Extracting archive

**Reply by kangzhaolong** [reply]
mysql

**Reply by kangzhaolong** [reply]
When I change the database with mysql client:
update luyacms4.cms_layout SET id=1 WHERE id=3
(there is only one item in my cms_layout table, and the id is 3 now).

Then every thing run well in the web page.

**Comment by nadar** [comment]
Can you find new database entries in your `cms_layout` after running import? Does that mine that each import creates a new entry in `cms_layout`? or can you confirm just the id is increased after running `import` command but still the same row?

**Reply by kangzhaolong** [reply]
No, no new entries.
Only the id is increased after running "import".

**Reply by kangzhaolong** [reply]
id is increased in the original record.

**Reply by nadar** [reply]
So this means, it deletes the layout, and creates a new one. Can you compare the `json_config` and `view_file` is there something changing?

**Comment by nadar** [comment]
Can you please post the FULL output log when running the command? It returns what happens:

for example:

```
< luya\cms\admin\importers\CmslayoutImporter:
< ╔═════╤═══════════════════════════════════════════════════╗
< ║ Key │ Value                                             ║
< ╟─────┼───────────────────────────────────────────────────╢
< ║ 0   │ cms layout importer finished with 2 layout files. ║
< ╚═════╧═══════════════════════════════════════════════════╝
```

**Reply by kangzhaolong** [reply]
.\vendor\bin\luya  import
LUYA import command (based on LUYA 2.2.0)

luya\cms\admin\importers\ThemeImporter:
╔═════╤════════════════════════════════════════╗
║ Key │ Value                                  ║
╟─────┼────────────────────────────────────────╢
║ 0   │ Theme importer finished with 0 themes. ║
╚═════╧════════════════════════════════════════╝

luya\cms\admin\importers\BlockImporter:
╔═════╤══════════════════════════════════════════════════════════════════════════════════════════╗
║ Key │ Value                                                                                    ║
╟─────┼──────────────────────────────────────────────────────────────────────────────────────────╢
║ 0   │ block \luya\cms\frontend\blocks\HtmlBlock: remains the same, nothing to update           ║
╟─────┼──────────────────────────────────────────────────────────────────────────────────────────╢
║ 1   │ block \luya\cms\frontend\blocks\MirrorLanguageBlock: remains the same, nothing to update ║
╟─────┼──────────────────────────────────────────────────────────────────────────────────────────╢
║ 2   │ block \luya\cms\frontend\blocks\ModuleBlock: remains the same, nothing to update         ║
╟─────┼──────────────────────────────────────────────────────────────────────────────────────────╢
║ 3   │ block \luya\bootstrap4\blocks\CarouselBlock: remains the same, nothing to update         ║
╟─────┼──────────────────────────────────────────────────────────────────────────────────────────╢
║ 4   │ block \luya\bootstrap4\blocks\ImageBlock: remains the same, nothing to update            ║
╟─────┼──────────────────────────────────────────────────────────────────────────────────────────╢
║ 5   │ block \luya\generic\blocks\FileListBlock: remains the same, nothing to update            ║
╟─────┼──────────────────────────────────────────────────────────────────────────────────────────╢
║ 6   │ block \luya\generic\blocks\LineBlock: remains the same, nothing to update                ║
╟─────┼──────────────────────────────────────────────────────────────────────────────────────────╢
║ 7   │ block \luya\generic\blocks\ListBlock: remains the same, nothing to update                ║
╟─────┼──────────────────────────────────────────────────────────────────────────────────────────╢
║ 8   │ block \luya\generic\blocks\QuoteBlock: remains the same, nothing to update               ║
╟─────┼──────────────────────────────────────────────────────────────────────────────────────────╢
║ 9   │ block \luya\generic\blocks\TextBlock: remains the same, nothing to update                ║
╟─────┼──────────────────────────────────────────────────────────────────────────────────────────╢
║ 10  │ block \luya\generic\blocks\TitleBlock: remains the same, nothing to update               ║
╟─────┼──────────────────────────────────────────────────────────────────────────────────────────╢
║ 11  │ block \luya\generic\blocks\WysiwygBlock: remains the same, nothing to update             ║
╟─────┼──────────────────────────────────────────────────────────────────────────────────────────╢
║ 12  │ block \luya\cms\frontend\blocks\HtmlBlock: remains the same, nothing to update           ║
╟─────┼──────────────────────────────────────────────────────────────────────────────────────────╢
║ 13  │ block \luya\cms\frontend\blocks\MirrorLanguageBlock: remains the same, nothing to update ║
╟─────┼──────────────────────────────────────────────────────────────────────────────────────────╢
║ 14  │ block \luya\cms\frontend\blocks\ModuleBlock: remains the same, nothing to update         ║
╟─────┼──────────────────────────────────────────────────────────────────────────────────────────╢
║ 15  │ Block importer finished with 15 blocks.                                                  ║
╚═════╧══════════════════════════════════════════════════════════════════════════════════════════╝

luya\cms\admin\importers\CmslayoutImporter:
╔═════╤═══════════════════════════════════════════════════╗
║ Key │ Value                                             ║
╟─────┼───────────────────────────────────────────────────╢
║ 1   │ cms layout importer finished with 1 layout files. ║
╚═════╧═══════════════════════════════════════════════════╝
Importer run successful.

**Reply by nadar** [reply]
So i don't think this is the problem, because it would say something like `New file XYZ found and registered.` or `Existing file XYZ updated.`. I think there is something wrong with your "database", or you have multiple connections or something like that. Sounds strange, but i don't think its a problem from the importer, we run them everywhere on multiple projects every day.

**Reply by nadar** [reply]
ps: you can test with latest dev-master, i commited a verysmall update regarding output (#https://github.com/luyadev/luya-module-cms/commit/a9c60a372756868e6cef59dd92979066e6d791c8) use `"luyadev/luya-module-cms":"dev-master as 4.3.0"`

**Reply by kangzhaolong** [reply]
OK, thanks, I check it again.

**Reply by kangzhaolong** [reply]
>>>I think there is something wrong with your "database", or you have multiple connections or something like that. 

I have two database source configured in my config file. Maybe that is the reason for this issue.
